components/form/form_validation.php

```diff
@@ -589,10 +589,18 @@ function set_type()
 
     function check(string $msg = null)
     {
-        if (!filter_var($this->value, FILTER_VALIDATE_EMAIL)) {
+        if ($this->should_test() && !filter_var($this->value, FILTER_VALIDATE_EMAIL)) {
             $this->set_error($msg ?? "Format d'email invalide");
         }
     }
+
+    function unity()
```

Comment:
```suggestion
    function unique()
```
It sounds cool, but it's not what it means 😅